components/org.wso2.carbon.identity.authenticator.risk/src/main/java/org/wso2/carbon/identity/authenticator/risk/internal/JavascriptFunctionsSupplierComponent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.authenticator.risk.internal;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.application.authentication.framework.JsFunctionRegistry;
+import org.wso2.carbon.identity.authenticator.risk.GetRiskScoreFunction;
+import org.wso2.carbon.identity.authenticator.risk.GetRiskScoreFunctionImpl;
+
+/**
+ * Register the java function which calculates the risk of auth requests
+ */
+@Component(
+        name = "identity.authenticator.risk.authenticator.risk.function.component",
+        immediate = true)
+public class JavascriptFunctionsSupplierComponent {
+
+    private JsFunctionRegistry jsFunctionRegistry;
+    private GetRiskScoreFunctionImpl getRiskScoreFunction;
+
+    @Activate
+    protected void activate(ComponentContext ctxt) {
+        getRiskScoreFunction = new GetRiskScoreFunctionImpl();
+        jsFunctionRegistry.register(JsFunctionRegistry.Subsystem.SEQUENCE_HANDLER, "getRiskScore",
+                (GetRiskScoreFunction) getRiskScoreFunction::getRiskScore);
```

Comment:
better to add a debug log here to indicate that component activated successfully.

Reply:
added